### PR TITLE
Allow GHC 9.10

### PR DESCRIPTION
**Pull request by ysangkok**
This is related to #1753, but adds a bunch of allow-newers. There are so many
dependencies of servant, that it will take a long time for all of them to
catch up. To make sure GHC 9.10 keeps working in this repo, let's test with
these allow-newers. Released packages won't have these bound overrides, of
course.

### CODE REVIEW

**Review by tchoutri — APPROVED** [review]

**Comment by ysangkok** [comment]
I thought the failure was because of the colors, but it still fails locally with this snippet added to `cabal.project`:

```
if(impl(ghc >= 9.10.1))                                                             
  package servant                                                                   
    ghc-options: -fdiagnostics-color=never
```

See [flag reference](https://downloads.haskell.org/ghc/latest/docs/users_guide/flags.html)

**Comment by ysangkok** [comment]
This is the error in case the CI logs get deleted:
```
src/Servant/API/TypeLevel.hs:270: failure in expression `instance AtMostOneFragment FailAPI'
expected: ...
          ...Only one Fragment allowed per endpoint in api...
          ...
          ...In the instance declaration for...
 but got: <interactive>:6[39](https://github.com/haskell-servant/servant/actions/runs/9387799345/job/25851582146#step:10:40):10: error: [GHC-64725]
              • Only one Fragment allowed per endpoint in api ‘Fragment Bool
                                                               :> (Fragment Int :> Get '[JSON] NoContent)’.
          .................................................................................O^
              • In the instance declaration for ‘AtMostOneFragment FailAPI’
```

File: cabal.project [hunk 1]
```diff
@@ -69,3 +69,29 @@ if(impl(ghc >= 9.6.1))
...
+allow-newer: postgresql-simple:base
+allow-newer: postgresql-simple:template-haskell
+allow-newer: postgresql-simple:containers
+allow-newer: postgresql-libpq:base
+allow-newer: servant-swagger-ui:base
+allow-newer: servant-swagger-ui-core:base
+allow-newer: file-embed-lzma:base
+allow-newer: file-embed-lzma:template-haskell
+allow-newer: file-embed-lzma:filepath
+allow-newer: lzma:base
```
Comment (ysangkok) [inline cabal.project:82]: This one was already bumped on hackage: https://hackage.haskell.org/package/lzma

File: cabal.project [hunk 2]
```diff
@@ -69,3 +69,29 @@ if(impl(ghc >= 9.6.1))
 
   package servant-server
     ghc-options: -fprint-redundant-promotion-ticks
+
+allow-newer: postgresql-simple:base
+allow-newer: postgresql-simple:template-haskell
+allow-newer: postgresql-simple:containers
+allow-newer: postgresql-libpq:base
```
Comment (ysangkok) [inline cabal.project:76]: blocked on 
* https://github.com/haskellari/postgresql-libpq/pull/59

File: cabal.project [hunk 3]
```diff
@@ -69,3 +69,29 @@ if(impl(ghc >= 9.6.1))
...
+allow-newer: lzma:base
+allow-newer: servant-openapi3:base
+allow-newer: insert-ordered-containers:base
+allow-newer: insert-ordered-containers:lens
+allow-newer: openapi3:base
+allow-newer: openapi3:template-haskell
+allow-newer: swagger2:base
+allow-newer: swagger2:template-haskell
+allow-newer: swagger2:lens
+allow-newer: swagger2:containers
```
Comment (ysangkok) [inline cabal.project:91]: blocked on

* https://github.com/GetShopTV/swagger2/issues/254

File: cabal.project [hunk 4]
```diff
@@ -69,3 +69,29 @@ if(impl(ghc >= 9.6.1))
...
+allow-newer: postgresql-libpq:base
+allow-newer: servant-swagger-ui:base
+allow-newer: servant-swagger-ui-core:base
+allow-newer: file-embed-lzma:base
+allow-newer: file-embed-lzma:template-haskell
+allow-newer: file-embed-lzma:filepath
+allow-newer: lzma:base
+allow-newer: servant-openapi3:base
+allow-newer: insert-ordered-containers:base
+allow-newer: insert-ordered-containers:lens
```
Comment (ysangkok) [inline cabal.project:85]: blocked on

* https://github.com/phadej/insert-ordered-containers/issues/60

**Pull request merged**